pex/environment.py

```diff
@@ -528,10 +561,19 @@ def resolve_dists(
         self,
         reqs,  # type: Iterable[Requirement]
         exclude_configuration=ExcludeConfiguration(),  # type: ExcludeConfiguration
+        result_type=None,  # type: Optional[InstallableType.Value]
     ):
         # type: (...) -> Iterable[FingerprintedDistribution]
 
-        self._update_candidate_distributions(self.iter_distributions())
+        result_type_wheel_file = False
+        if result_type is not None:
+            from pex.pep_427 import InstallableType
```

Comment:
I'm curious, why is this import conditional?

Reply:
Because Python is hella slow and this is in the boot hot path. Normal imports in this file add ~50ms to boot overhead.